src/TLuaInterpreter.cpp

```diff
@@ -17197,6 +17198,110 @@ int TLuaInterpreter::getMouseEvents(lua_State * L)
     return 1;
 }
 
+int TLuaInterpreter::setConfig(lua_State * L)
+{
+    auto& host = getHostFromLua(L);
+    QString key = getVerifiedString(L, __func__, 1, "key");
+    if (key.isEmpty()) {
+        return warnArgumentValue(L, __func__, "you must provide key");
+    }
+
+    auto success = [&]()
+    {
+        if (mudlet::debugMode) {
+            TDebug(Qt::white, Qt::blue) << qsl("setConfig: a script has changed %1\n").arg(key) >> &host;
+        }
+        lua_pushboolean(L, true);
+        return 1;
+    };
+
+    if (host.mpMap && host.mpMap->mpMapper) {
+        if (key == qsl("roomSize")) {
+            host.mpMap->mpMapper->slot_setRoomSize(getVerifiedInt(L, __func__, 2, "value"));
+            return success();
+        }
+        if (key == qsl("exitSize")) {
+            host.mpMap->mpMapper->slot_setExitSize(getVerifiedInt(L, __func__, 2, "value"));
+            return success();
+        }
+        if (key == qsl("roundRooms")) {
+            host.mpMap->mpMapper->slot_toggleRoundRooms(getVerifiedBool(L, __func__, 2, "value"));
+            return success();
+        }
+        if (key == qsl("showRoomIds")) {
+            host.mpMap->mpMapper->slot_setShowRoomIds(getVerifiedBool(L, __func__, 2, "value"));
+            return success();
+        }
+        if (key == qsl("showMapInfo")) {
+            host.mMapInfoContributors.insert(getVerifiedString(L, __func__, 2, "value"));
+            host.mpMap->mpMapper->slot_updateInfoContributors();
+            return success();
+        }
+        if (key == qsl("hideMapInfo")) {
+            host.mMapInfoContributors.remove(getVerifiedString(L, __func__, 2, "value"));
+            host.mpMap->mpMapper->slot_updateInfoContributors();
+            return success();
+        }
+#if defined(INCLUDE_3DMAPPER)
+        if (key == qsl("show3dView")) {
+            host.mpMap->mpMapper->slot_toggle3DView(getVerifiedBool(L, __func__, 2, "value"));
+            return success();
+        }
+#endif
+        if (key == qsl("panelVisible")) {
```

Comment:
I just noticed... This option name doesn't really convey which panel is set visible/invisible. Can we possibly nest the mapper options somehow? There are others that are not 100% clear to me either.

Reply:
We can for sure. I'd also like to keep them as short as it is reasonably feasible at the same time, this is what I am working off:

![image](https://user-images.githubusercontent.com/110988/165461750-cbfeef9d-c2d0-47cd-9d3c-2d33a420b610.png)

So ideally we don't make everything very lengthy, but just descriptive enough to be understood. What's your take on that?

Reply:
I think those in general are fine. But the mapper options lose a bit of context when not being shown on the `Mapper` tab. Especially the `show3dView` is problematic, but the others could be a little confusing as well. That's why I had the nesting idea. Maybe put `map` into those option names where they are not yet?

- `roomSize` - `mapRoomSize`
- `exitSize` - `mapExitSize`
- `roundRooms` - `mapRoundRooms` (I don't like this one, but I don't have a better idea)
- `showRoomIds` - `showRoomIdsOnMap` (breaks with the map prefix, but the other one feels strange)
- `showMapInfo` keep
- `show3dView` - `show3dMapView`
- `showRoomBorders` - `mapShowRoomBorders`

Reply:
Great idea - applied them all.

Reply:
yeah but mapX is more in keeping with inputLineX, mainDisplayX and specialX token names.

I'd honestly even suggest `mapShowRoomIds` as the way to go there, it's no more awkward than `mainDisplayFixUnnecessaryLinebreaks` and the consistency is I think more important in this case.

Reply:
Yeah, I agree with @demonnic